tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,283 @@
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_edit(remove_resource):
+    client = get_user_client()
+    cloud_credential = cloud_credential_do(client)
+    # User1 should be able to edit own credential
+    do_cloud_credential_config = {"accessToken": "12345"}
+    client.update(cloud_credential,
+                  digitaloceancredentialConfig=do_cloud_credential_config)
+    # Admin should be able to edit credential created by user1
+    admin_client = get_admin_client()
+    do_cloud_credential_config = {"accessToken": "678910"}
+    admin_client.update(cloud_credential,
+                  digitaloceancredentialConfig=do_cloud_credential_config)
+    # User 2 should not be able to edit credential created by user1
+    user1, user1_token = create_user(get_admin_client())
+    user1_client = get_client_for_token(user1_token)
+    with pytest.raises(ApiError) as e:
+        do_cloud_credential_config = {"accessToken": "111213"}
+        user1_client.update(cloud_credential,
+                            digitaloceancredentialConfig=
+                            do_cloud_credential_config)
+    assert e.value.error.status == 403
+    time.sleep(.5)
+    remove_resource(cloud_credential)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_deploy_cluster(remove_resource):
+    admin_client = get_admin_client()
+    cloud_credential = cloud_credential_do(admin_client)
+    node_template = node_template_do(admin_client, cloud_credential)
+    cluster = admin_client.create_cluster(node_template)
+    node_name = random_name()
+    node = [{"hostnamePrefix": node_name,
+             "nodeTemplateId": node_template.id,
+             "requestedHostname": node_name,
+             "etcd": True,
+             "controlPlane": True,
+             "worker": True,
+             "quantity": 1}]
+    rancherKubernetesEngineConfig = rke_config
+    cluster, node_pools = create_and_validate_cluster(
+        admin_client, node, rancherKubernetesEngineConfig)
+    cluster_cleanup(admin_client, cluster)
+
+
+##############################################################################
+
+
+def node_template_do(client, cloud_credential=None):
+    if cloud_credential:
+        do_cloud_credential_config = cloud_credential
+    else:
+        do_cloud_credential_config = {"accessToken": DO_ACCESSKEY}
+    do_cloud_credential = client.create_cloud_credential(
+        digitaloceancredentialConfig=do_cloud_credential_config
+    )
+    node_template = client.create_node_template(
+        digitaloceanConfig={"region": "nyc3",
+                            "size": "2gb",
+                            "image": "ubuntu-18-04-x64"},
+        name=random_name(),
+        driver="digitalocean",
+        cloudCredentialId=do_cloud_credential.id,
+        engineInstallURL=engine_install_url,
+        useInternalIpAddress=True)
+    node_template = client.wait_success(node_template)
+    return node_template
+
+
+def cloud_credential_do(client):
```

Comment:
can we rename this to create_cloud_credential_do() to make it more readable ?

Reply:
Method has been renamed.